pkg/frontend/admin_vm_actions.go

```diff
@@ -0,0 +1,60 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"net/http"
+	"strings"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/database/cosmosdb"
+	"github.com/Azure/ARO-RP/pkg/frontend/adminactions"
+)
+
+type azVmActionsWrapper struct {
+	vmName       string
+	adminActions adminactions.AzureActions
```

Comment:
Would it be beneficial to define an `AzureVmActions` interface so we could ensure that only VM Actions are being included in this wrapper?

```go
type AzureVmActions interface {
    VMRedeployAndWait(ctx context.Context, vmName string) error
	VMStartAndWait(ctx context.Context, vmName string) error
	VMStopAndWait(ctx context.Context, vmName string) error
	VMSizeList(ctx context.Context) ([]mgmtcompute.ResourceSku, error)
	VMResize(ctx context.Context, vmName string, vmSize string) error
	VMSerialConsole(ctx context.Context, w http.ResponseWriter, log *logrus.Entry, vmName string) error
}
```

We could also use that to compose the `AzureActions` interface

```go
type AzureActions interface {
    ResourcesList(ctx context.Context) ([]byte, error)
	NICReconcileFailedState(ctx context.Context, nicName string) error
    AzureVmActions
}
```

Reply:
Seems a unnecessary to me but I am happy to get more input. @bennerv @m1kola